### PR TITLE
Fixed bug: Disconnected domain (no access token present) is unnecessarily hitting /api/v1/user/friends

### PR DESCRIPTION
**Pull request by utkarshgautamnyu**
Refer to Fogbugz case: https://highfidelity.fogbugz.com/f/cases/6470/Disconnected-domain-no-access-token-present-is-unnecessarily-hitting-api-v1-user-friends

Made sendRequest from  getDomainOwnerFriendsList depend on hasValidAccessToken being true 

Testing Notes:
1) Run the built PR
2) Run the domain server
3) Check domain server logs, following sample line should **NOT** be there:
[07/21 20:24:04] [DEBUG] [hifi.networking] No valid access token present. Bailing on invoked request to "api/v1/user/friends" that requires authentication

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1895/HighFidelity-Beta-PR11035-884bdb3b7e776a21f2216ccd6e248c2e73a7b1b5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1895/HighFidelity-Beta-PR11035-884bdb3b7e776a21f2216ccd6e248c2e73a7b1b5.exe

**Comment by Nex-Pro** [comment]
Testing this.

**Comment by Nex-Pro** [comment]
### PASS

@utkarshgautamnyu I ran this domain server build and entered the domain logged in on one of my accounts. I then checked the domain logs for bail messages containing api/v1/user/friends
There were none. Please let me know if this is satisfactory.

**Comment by utkarshgautamnyu** [comment]
@Nex-Pro : yes this works, thanks!

File: domain-server/src/DomainGatekeeper.cpp [hunk 1]
```diff
@@ -927,9 +927,13 @@ void DomainGatekeeper::getDomainOwnerFriendsList() {
     callbackParams.errorCallbackMethod = "getDomainOwnerFriendsListErrorCallback";
 
     const QString GET_FRIENDS_LIST_PATH = "api/v1/user/friends";
-    DependencyManager::get<AccountManager>()->sendRequest(GET_FRIENDS_LIST_PATH, AccountManagerAuth::Required,
-                                                          QNetworkAccessManager::GetOperation, callbackParams, QByteArray(),
-                                                          NULL, QVariantMap());
+    if (DependencyManager::get<AccountManager>()->hasValidAccessToken())
+    {
```
Comment (AndrewMeadows) [inline domain-server/src/DomainGatekeeper.cpp]: As per the coding standard, put the opening brace `{` on the same line as the `if`.

**Comment by utkarshgautamnyu** [comment]
@AndrewMeadows I have made the requested changes

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1906/HighFidelity-Beta-PR11035-5771ee27e1f510a563b4d25b39f8071d3e93d547.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1906/HighFidelity-Beta-PR11035-5771ee27e1f510a563b4d25b39f8071d3e93d547.exe

**Pull request merged**